### PR TITLE
GEM-110 | add DELETE /external/expenses/{expense_id}/groups/{group_id}

### PR DESCRIPTION
**Pull request by pwgacek**

### CODE REVIEW

File: src/integration/kotlin/pl/edu/agh/gem/integration/controller/ExpenseControllerIT.kt [hunk 1]
```diff
@@ -536,5 +539,67 @@ class ExpenseControllerIT(
...
+            // when
+            val response = service.delete(createGemUser(USER_ID, EMAIL), GROUP_ID, EXPENSE_ID)
+
+            // then
+            response shouldHaveHttpStatus OK
+            repository.findByExpenseIdAndGroupId(EXPENSE_ID, GROUP_ID).also {
+                it.shouldBeNull()
+            }
+        }
+        should("return forbidden if user is not a group member") {
```
Comment (lawos98) [inline src/integration/kotlin/pl/edu/agh/gem/integration/controller/ExpenseControllerIT.kt:558]: why we check if use user is group meber when only creator can delete that ?

File: src/main/kotlin/pl/edu/agh/gem/external/controller/ApiExceptionHandler.kt [hunk 2]
```diff
@@ -39,6 +40,13 @@ class ApiExceptionHandler {
         return ResponseEntity(handleError(exception), FORBIDDEN)
     }
 
+    @ExceptionHandler(UserNotExpenseCreatorException::class)
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/external/controller/ApiExceptionHandler.kt:43]: maybe more like `ExpenseDeletionAcessException` ?

File: src/main/kotlin/pl/edu/agh/gem/external/persistence/MongoExpenseRepository.kt [hunk 3]
```diff
@@ -32,6 +32,10 @@ class MongoExpenseRepository(
         mongo.save(expense.toEntity())
     }
 
+    override fun delete(expense: Expense) {
+        mongo.remove(expense.toEntity())
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/external/persistence/MongoExpenseRepository.kt:36]: maybe use one operation isdead of two ( findOneAndDelete() )

File: src/main/kotlin/pl/edu/agh/gem/internal/service/ExpenseService.kt [hunk 4]
```diff
@@ -116,10 +116,25 @@ class ExpenseService(
...
     }
+
+    fun deleteExpense(expenseId: String, groupId: String, userId: String) {
+        val expenseToDelete = expenseRepository.findByExpenseIdAndGroupId(expenseId, groupId) ?: throw MissingExpenseException(expenseId, groupId)
+
+        if (!userId.isCreator(expenseToDelete)) {
+            throw UserNotExpenseCreatorException(userId, expenseId)
+        }
+
+        expenseRepository.delete(expenseToDelete)
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/internal/service/ExpenseService.kt:127]: dont we want to add new collection to archive history ?

File: src/main/kotlin/pl/edu/agh/gem/external/persistence/MongoExpenseRepository.kt [hunk 5]
```diff
@@ -32,6 +32,10 @@ class MongoExpenseRepository(
         mongo.save(expense.toEntity())
     }
 
+    override fun delete(expense: Expense) {
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/external/persistence/MongoExpenseRepository.kt:35]: could you also add integration tests for that method ?

File: src/integration/kotlin/pl/edu/agh/gem/integration/controller/ExpenseControllerIT.kt (hunk 1 above)
Comment (pwgacek) [reply src/integration/kotlin/pl/edu/agh/gem/integration/controller/ExpenseControllerIT.kt:558]: as discussed on daily. This way we will not share the information if expense with given id exists or not

**Review by lawos98 — APPROVED** [review]

**Pull request merged**